cpp/src/arrow/compute/kernels/aggregate_internal.h

```diff
@@ -132,7 +142,13 @@ struct GetSumType<T, enable_if_integer<T>> {
 };
 
 template <typename T>
-struct GetSumType<T, enable_if_decimal<T>> {
+struct GetSumType<T, enable_if_decimal32<T>> {
+  using SumType = arrow::Decimal64;
```

Comment:
Why? This makes things slightly irregular and more difficult to predict for the user.

Reply:
See discussion [here](https://github.com/apache/arrow/pull/43957#discussion_r1750962486) it's to avoid overflows, we do the same for the sumtype for other integer types, which all have a Sum type of 64-bit integers

Reply:
Decimal is **not** an integer type. The fact that it's coded as an integer is irrelevant.

Reply:
that's a good point and I agree, i'll put it back.